### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by AnnaGolosova**

### CODE REVIEW

**Review by LBHSPreston — COMMENTED** [review]
I've just enabled CircleCi for this project and there are a number of items that are failing which would prevent the deployment. https://app.circleci.com/pipelines/github/LBHackney-IT/Financial-Summary

**Review by LBHSPreston — DISMISSED** [review]

**Review by sandipchandra — COMMENTED** [review]
All looks good.

**Comment by LBHSPreston** [comment]
Looks like there are a few checks failing now.  We can meet up to review if necessary.

File: FinancialSummaryApi/V1/Boundary/Response/WeeklySummaryResponse.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
+using System;
+
+namespace FinancialSummaryApi.V1.Boundary.Response
+{
+    public class WeeklySummaryResponse
+    {
+        /// <summary>
```
Comment (AnnaGolosova) [inline FinancialSummaryApi/V1/Boundary/Response/WeeklySummaryResponse.cs:7]: @sandipchandra I think here should be <example> attributes

File: FinancialSummaryApi/V1/Boundary/Request/AddWeeklySummaryRequest.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
+using FinancialSummaryApi.V1.Infrastructure;
+using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace FinancialSummaryApi.V1.Boundary.Request
+{
+    public class AddWeeklySummaryRequest
+    {
+        public Guid Id { get; set; }
```
Comment (AnnaGolosova) [inline FinancialSummaryApi/V1/Boundary/Request/AddWeeklySummaryRequest.cs:9]: @sandipchandra We don't need id in POST method. We should create new id on our side

File: FinancialSummaryApi/V1/Controllers/WeeklySummaryController.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+        /// <summary>
+        /// Get Weekly summary model by provided Id
+        /// </summary>
+        /// <param name="id">The value by which we are looking for an weekly summary</param>
+        /// <response code="200">Success. Weekly summary models is saved successfully</response>
+        /// <response code="400">Bad Request</response>
+        /// <response code="404">Weekly Summary with provided id cannot be found</response>
+        /// <response code="500">Internal Server Error</response>
+        [ProducesResponseType(typeof(AssetSummaryResponse), StatusCodes.Status200OK)]
```
Comment (AnnaGolosova) [inline FinancialSummaryApi/V1/Controllers/WeeklySummaryController.cs:39]: @sandipchandra WeeklySummaryResponse? :)

File: FinancialSummaryApi/V1/Controllers/WeeklySummaryController.cs [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        /// Get Weekly Transaction summary for a given date range
+        /// </summary>
+        /// <param name="targetId">The date when the requested data was generated</param>
+        /// <param name="startDate">The value by which we are looking for an weekly summary</param>
+        /// <param name="endDate">The value by which we are looking for an weekly summary</param>
+        /// <response code="200">Success. Weekly summary models was received successfully</response>
+        /// <response code="400">Bad Request</response>
+        /// <response code="404">Weekly Summary with provided id cannot be found</response>
+        /// <response code="500">Internal Server Error</response>
+        [ProducesResponseType(typeof(WeeklySummaryResponse), StatusCodes.Status200OK)]
```
Comment (AnnaGolosova) [inline FinancialSummaryApi/V1/Controllers/WeeklySummaryController.cs:67]: typeof(List<WeeklySummaryResponse>)

File: FinancialSummaryApi/V1/Controllers/WeeklySummaryController.cs [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        [ProducesResponseType(typeof(BaseErrorResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(BaseErrorResponse), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(typeof(BaseErrorResponse), StatusCodes.Status500InternalServerError)]
+        [HttpGet]
+        [Route("{id}")]
+        public async Task<IActionResult> Get([FromRoute] Guid id)
+        {
+            var assetSummary = await _getWeeklySummaryByIdUseCase.ExecuteAsync(id).ConfigureAwait(false);
+
+            if (assetSummary == null)
```
Comment (AnnaGolosova) [inline FinancialSummaryApi/V1/Controllers/WeeklySummaryController.cs:49]: I think it's copy-past issue. We don't need to use `assetSummary` variable here

File: FinancialSummaryApi/V1/Controllers/WeeklySummaryController.cs [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+                return NotFound(new BaseErrorResponse((int) HttpStatusCode.NotFound, "Weekly Summary by provided Id not found!"));
+            }
+
+            return Ok(assetSummary);
+        }
+
+        /// <summary>
+        /// Get Weekly Transaction summary for a given date range
+        /// </summary>
+        /// <param name="targetId">The date when the requested data was generated</param>
```
Comment (AnnaGolosova) [inline FinancialSummaryApi/V1/Controllers/WeeklySummaryController.cs:60]: Something is wrong with documentation

File: FinancialSummaryApi/V1/Controllers/WeeklySummaryController.cs [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+        }
+
+        /// <summary>
+        /// Create new Weekly summary model
+        /// </summary>
+        /// <param name="weeklySummary">Weekly summary model for create</param>
+        /// <response code="201">Created. Weekly summary model was created successfully</response>
+        /// <response code="400">Bad Request</response>
+        /// <response code="500">Internal Server Error</response>
+        [ProducesResponseType(typeof(AssetSummaryResponse), StatusCodes.Status201Created)]
```
Comment (AnnaGolosova) [inline FinancialSummaryApi/V1/Controllers/WeeklySummaryController.cs:91]: WeeklySummaryResponse

File: FinancialSummaryApi/V1/Factories/WeeklySummaryFactory.cs [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    {
+        public static WeeklySummary ToWeeklySummaryDomain(this WeeklySummaryDbEntity databaseEntity)
+        {
+            if (databaseEntity == null)
+            {
+                return null;
+            }
+            if (databaseEntity.Id == null)
+            {
+                throw new Exception("Loaded data from the database cannot be parsed as a Asset data. Id: " + databaseEntity.Id);
```
Comment (AnnaGolosova) [inline FinancialSummaryApi/V1/Factories/WeeklySummaryFactory.cs:18]: What does this error message means?

File: FinancialSummaryApi/V1/Gateways/DynamoDbContextWrapper.cs [hunk 9]
```diff
@@ -11,5 +12,13 @@ public virtual Task<List<FinanceSummaryDbEntity>> ScanAsync(IDynamoDBContext con
         {
             return context.ScanAsync<FinanceSummaryDbEntity>(conditions, operationConfig).GetRemainingAsync();
         }
+        public virtual Task<List<WeeklySummaryDbEntity>> ScanSummaryAsync(IDynamoDBContext context, IEnumerable<ScanCondition> conditions, DynamoDBOperationConfig operationConfig = null)
```
Comment (AnnaGolosova) [inline FinancialSummaryApi/V1/Gateways/DynamoDbContextWrapper.cs:14]: I think we can change `ScanAsync` method to be generic

File: FinancialSummaryApi/V1/Gateways/DynamoDbGateway.cs [hunk 10]
```diff
@@ -96,7 +96,43 @@ public async Task<RentGroupSummary> GetRentGroupSummaryByNameAsync(string rentGr
...
+
+        public async Task AddAsync(WeeklySummary weeklySummary)
+        {
+            await _dynamoDbContext.SaveAsync(weeklySummary.ToDatabase()).ConfigureAwait(false);
+        }
+        public async Task<WeeklySummary> GetWeeklySummaryByIdAsync(Guid id)
+        {
+            var scanConditions = new List<ScanCondition>();
+
+            scanConditions.Add(new ScanCondition("Id", ScanOperator.Equal, id));
```
Comment (AnnaGolosova) [inline FinancialSummaryApi/V1/Gateways/DynamoDbGateway.cs:125]: We don't use `scanConditions` variable here

File: FinancialSummaryApi/V1/Gateways/DynamoDbGateway.cs [hunk 11]
```diff
@@ -96,7 +96,43 @@ public async Task<RentGroupSummary> GetRentGroupSummaryByNameAsync(string rentGr
...
+
+            var data = await _wrapper.LoadSummaryAsync(_dynamoDbContext, id).ConfigureAwait(false);
+
+            return data?.ToWeeklySummaryDomain();
+        }
+        #endregion
+
         private static Tuple<DateTime, DateTime> GetDayRange(DateTime date)
             => new Tuple<DateTime, DateTime>(date.Date, date.Date.AddHours(23).AddMinutes(59));
+
```
Comment (AnnaGolosova) [inline FinancialSummaryApi/V1/Gateways/DynamoDbGateway.cs:135]: Do we need these enters here?

File: FinancialSummaryApi/V1/Controllers/WeeklySummaryController.cs [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+        /// <response code="200">Success. Weekly summary models was received successfully</response>
+        /// <response code="400">Bad Request</response>
+        /// <response code="404">Weekly Summary with provided id cannot be found</response>
+        /// <response code="500">Internal Server Error</response>
+        [ProducesResponseType(typeof(WeeklySummaryResponse), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(BaseErrorResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(BaseErrorResponse), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(typeof(BaseErrorResponse), StatusCodes.Status500InternalServerError)]
+        [HttpGet]
+        public async Task<IActionResult> GetAll( [FromQuery] Guid targetId, [FromQuery] string startDate, [FromQuery] string endDate)
```
Comment (AnnaGolosova) [inline FinancialSummaryApi/V1/Controllers/WeeklySummaryController.cs:72]: We don't we accept DateTime `startDate` and `endDate` params here?

**Pull request merged**